src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -168,6 +168,17 @@ theorem getMsbD_eq_getMsb?_getD (x : BitVec w) (i : Nat) :
       intros
       omega
 
+theorem eq_of_getElem_eq {x y : BitVec w}
+    (pred : ∀ (i : Fin w), x[i] = y[i]) : x = y := by
```

Comment:
I think the preferred spelling of this would be to use `Nat`  instead of `Fin` to be consistent with the bits of the API added in #5200. 
```suggestion
    (pred : ∀ (i : Nat) (_ : i < w), x[i] = y[i]) : x = y := by
```
See also the analogous lemma [`List.ext_getElem`](https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Lemmas.html#List.ext_getElem).

Reply:
It seems for `BitVec` we used historically:

```lean
theorem eq_of_getLsbD_eq {x y : BitVec w}
    (pred : ∀(i : Fin w), x.getLsbD i.val = y.getLsbD i.val) : x = y := by

theorem eq_of_getMsbD_eq {x y : BitVec w}
    (pred : ∀(i : Fin w), x.getMsbD i.val = y.getMsbD i.val) : x = y := by
```

I guess we should at the very least remain consistent within `BitVec`.

I now added both variants to this PR and ported one theorem to show the code impact. Happy to flip to Nat, but would appreciate a comment why this is preferable over Fin.